astropy/wcs/src/wcslib_wrap.c

```diff
@@ -181,6 +181,7 @@ static PyWcsprm*
 PyWcsprm_cnew(void) {
   PyWcsprm* self;
   self = (PyWcsprm*)(&PyWcsprmType)->tp_alloc(&PyWcsprmType, 0);
+  memset(&self->x, 0, sizeof(struct wcsprm));
```

Comment:
`tp_alloc` initializes the memory with zeros:

>  It should return a pointer to a block of memory of adequate length for the instance, suitably aligned, and initialized to zeros[...]

So `self->x` should be `NULL`, shouldn't it? And I thought `memset` would be undefined behavior if *dest* is NULL.

However this is just educated guess, I haven't looked into this in detail.

Reply:
Ah, yes. It was in my "C equivalent" (that excludes Python from the equation) that the block was not initialized.